lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2709,8 +2716,7 @@ void ByteCodeGenerator::EmitDefaultArgs(FuncInfo *funcInfo, ParseNodeFnc *pnodeF
         m_writer.RecordCrossFrameEntryExitRecord(/* isEnterBlock = */ true);
         m_writer.Br(Js::OpCode::TryCatch, catchLabel);
 
-        // Rest cannot have a default argument, so we ignore it.
-        MapFormalsWithoutRest(pnodeFnc, emitDefaultArg);
+        MapFormals(pnodeFnc, emitDefaultArg);
```

Comment:
When the rest param is a pattern, we need to run `emitDefaultArg` on it in order to emit destructuring code.

When the rest param is not a pattern, there is no change to the existing bytecode output; `emitDefaultArg` will perform

- `sym->SetNeedDeclaration(false)`, which was happening below
- `EmitPropStore`, which is a no-op for this case